### PR TITLE
[workload] Fix the SDK band version and base the WBT on the sdk flow

### PR DESCRIPTION
**Pull request by lewing**

### CODE REVIEW

**Comment by akoeplinger** [comment]
hmm I'm not sure this works without some of the other changes from the NET10 PR

**Comment by lewing** [comment]
> hmm I'm not sure this works without some of the other changes from the NET10 PR

It doesn't work with them either, I trying to illustrate the problem to @marcpopMSFT

**Comment by lewing** [comment]
> > hmm I'm not sure this works without some of the other changes from the NET10 PR
> 
> It doesn't work with them either, I was trying to illustrate the problem in [dotnet/sdk#43070](https://github.com/dotnet/sdk/pull/43070) to @marcpopMSFT

Just to make it more clear the SDK band version we're constructing right now in main is `9.0.0-alpha.1` which is very wrong, but using `10.0.0-alpha.1` won't work properly in a 9.0 sdk (even if we alter wbt to work).  This PR now changes it to `9.0.0-rc.1` which should work here and will probably work in the SDK, at least sort of.

We might be able to reenable the multiple band logic and produce a 10.0.0-aplha.1 and a 9.0.0-rc.1 until the flow resolves?

**Comment by marcpopMSFT** [comment]
Why can't you make it `10.0.0-alpha.1` like it's supposed to be? We should probably have already updated the stage0 SDK in the sdk PR to a 10.0.100 version which would make the workloads from runtime work, wouldn't it? Any reason we can't update stage zero sdk there?

**Comment by lewing** [comment]
> Why can't you make it `10.0.0-alpha.1` like it's supposed to be? We should probably have already updated the stage0 SDK in the sdk PR to a 10.0.100 version which would make the workloads from runtime work, wouldn't it? Any reason we can't update stage zero sdk there?

Trying again with a recent 10.0.100 sdk

**Comment by lewing** [comment]
> > Why can't you make it `10.0.0-alpha.1` like it's supposed to be? We should probably have already updated the stage0 SDK in the sdk PR to a 10.0.100 version which would make the workloads from runtime work, wouldn't it? Any reason we can't update stage zero sdk there?
> 
> Trying again with a recent 10.0.100 sdk

@marcpopMSFT  current failure is that the alpha.2 SDK only has workload sets which breaks our test setup

**Comment by marcpopMSFT** [comment]
> @marcpopMSFT current failure is that the alpha.2 SDK only has workload sets which breaks our test setup
It has workload sets? You mean the baseline workload set is breaking the test setup somehow? CC @dsplaisted if that's the case to see if he has ideas on how we reversion in the workload set world.

File: src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs [hunk 1]
```diff
@@ -288,6 +288,12 @@ private bool InstallWorkloadManifest(ITaskItem workloadId, string name, string v
 
             string outputDir = FindSubDirIgnoringCase(manifestVersionBandDir, name);
 
+            if (!Directory.Exists(outputDir))
+            {
+                Log.LogError($"Could not find {name} directory at {outputDir}. Creating it..");
```
Comment (lewing) [inline src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs:293]:
```suggestion
                Log.LogMessage($"Could not find {name} directory at {outputDir}. Creating it..");
```

**Comment by lewing** [comment]
> > @marcpopMSFT current failure is that the alpha.2 SDK only has workload sets which breaks our test setup
> 
> It has workload sets? You mean the baseline workload set is breaking the test setup somehow? CC @dsplaisted if that's the case to see if he has ideas on how we reversion in the workload set world.

So far it was just expecting a directory that wasn't there.  Hopefully I was just overly pessimistic, will see if I can get it working here.

**Comment by marcpopMSFT** [comment]
Any luck? What folder? Do we have an option of having wasm workloads broken for a short time in sdk so we can get the rest of the repo and net10 work working?

**Comment by lewing** [comment]
> Any luck? What folder? Do we have an option of having wasm workloads broken for a short time in sdk so we can get the rest of the repo and net10 work working?

I had to flow some updates from emsdk which is not fast (see https://github.com/dotnet/dnceng/issues/4111 ) but I'm trying again now

File: eng/testing/workloads-testing.targets [hunk 2]
```diff
@@ -16,9 +16,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <!-- Required for running apps built with 9.0 sdk, but the sdk does
-         not yet support *running* with 9.0 sdk -->
-    <AdditionalSharedFrameworkToInstallArguments Include="-Version latest -Channel 8.0 -Quality daily" />
+    <!-- Required for running apps built with 10.0 sdk, but the sdk does
```
Comment (radical) [inline eng/testing/workloads-testing.targets:19]: Changes in this file look good.

File: src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs [hunk 3]
```diff
@@ -288,6 +288,12 @@ private bool InstallWorkloadManifest(ITaskItem workloadId, string name, string v
 
             string outputDir = FindSubDirIgnoringCase(manifestVersionBandDir, name);
 
+            if (!Directory.Exists(outputDir))
+            {
+                Log.LogMessage($"Could not find {name} directory at {outputDir}. Creating it..");
+                Directory.CreateDirectory(outputDir);
+            }
+
```
Comment (radical) [inline src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs:296]: Changes in this file look good.

**Review by steveisok — APPROVED** [review]

**Review by radical — APPROVED** [review]
The workload task+installation specific changes LGTM 👍

**Comment by lewing** [comment]
I've merged what was remaining of this into https://github.com/dotnet/runtime/pull/106599 which farther along now

**Pull request closed without merging**